austation/code/modules/power/fuel_rods.dm

```diff
@@ -1,146 +1,291 @@
 /obj/item/twohanded/required/fuel_rod
-	var/conversion = "plutonium"// what does this rod convert to when depleted?
 	var/rad_strength = 500
 	var/half_life = 2000 // how many depletion ticks are needed to half the fuel_power (1 tick = 1 second)
 	var/time_created = 0
 	var/og_fuel_power = 0.20 //the original fuel power value
-	var/process =  FALSE
+	var/process = FALSE
+	// The depletion where Final_Depletion() will be called (and does something)
+	var/depletion_threshold = 100
+	// How fast this rod will deplete
+	var/depletion_speed_modifier = 1
+	var/depleted_final = FALSE // Final_Depletion should run only once
 
-
-	// TC rod only vars (yes I know, slight shitcode having them stored here but I don't want to make another proc for tc rods)
-	var/grown = FALSE // has the rod fissiled enough for us to remove the grown TC?
-	var/percentage = 0 //progress towards tc transmutation in percentage
-
-/obj/item/twohanded/required/fuel_rod/New()
-	..()
+/obj/item/twohanded/required/fuel_rod/Initialize()
+	. = ..()
 	time_created = world.time
-	AddComponent(/datum/component/radioactive, rad_strength, src)
+	AddComponent(/datum/component/radioactive, rad_strength, src) // This should be temporary for it won't make rads go lower than 350
 	if(process)
 		START_PROCESSING(SSobj, src)
-	. = ..()
 
 /obj/item/twohanded/required/fuel_rod/Destroy()
 	if(process)
 		STOP_PROCESSING(SSobj, src)
 	. = ..()
 
+// Child types should override this or your fuel rod will be turned to plutonium fuel rod
+/obj/item/twohanded/required/fuel_rod/proc/Final_Depletion()
```

Comment:
procs should be lowercase except in very special cases. please lowercase your proc name.